KSP2Runtime/KontrolModuleExtensions.cs

```diff
@@ -35,11 +67,61 @@ public static class KontrolModuleExtensions {
             }
         }
 
+        public static EntrypointArgumentDescriptor[] GetFlightEntrypointArgumentDescriptors(this IKontrolModule module, ITO2Logger logger = null) => GetEntrypointParameterDescriptors(module, MainFlight, logger);
+        public static EntrypointArgumentDescriptor[] GetKSCEntrypointArgumentDescriptors(this IKontrolModule module, ITO2Logger logger = null) => GetEntrypointParameterDescriptors(module, MainKsc, logger);
+        public static EntrypointArgumentDescriptor[] GetTrackingEntrypointArgumentDescriptors(this IKontrolModule module, ITO2Logger logger = null) => GetEntrypointParameterDescriptors(module, MainTracking, logger);
+        public static EntrypointArgumentDescriptor[] GetEditorEntrypointArgumentDescriptors(this IKontrolModule module, ITO2Logger logger = null) => GetEntrypointParameterDescriptors(module, MainEditor, logger);
+
+        private static EntrypointArgumentDescriptor[] GetEntrypointParameterDescriptors(IKontrolModule module, string name, ITO2Logger logger = null) {
+            try {
+                IKontrolFunction function = module.FindFunction(name);
+                if (function == null || function.Parameters.Count <= 1) {
+                    throw new Exception($"Function {name} does not exist or does not have any parameters");
+                }
+
+                return function.Parameters.Skip(1).Select(param => {
+                    if (param.type.Name == "int") {
+                        var defaultValue = param.defaultValue as IntDefaultValue;
+                        if (defaultValue == null) {
+                            throw new Exception($"Parameter {param.name} does not have a default value");
```

Comment:
Instead of an exception I would go with a fallback value here (a default for the default so to speak).
I.e. if not otherwise specified in the code int = 0, float = 0.0, bool = false